mod/CheckHintData.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using Archipelago.MultiClient.Net.Enums;
+
+namespace ArchipelagoRandomizer
+{
+    public class CheckHintData : MonoBehaviour
+    {
+        // Since arcs have no idea which logs they're connected to,
+        // we record them and their other location checks here in a list
+        // so we can easily see them in Unity Explorer
+        // Currently unused in any code but may use it later
+        public List<Location> Locations = new List<Location>();
+        public CheckImportance Importance = CheckImportance.Junk;
+        public bool HasBeenFound = false;
```

Comment:
If I'm understanding this right, Importance and HasBeenFound are never accessed outside of this class, and are only used in NomaiWallColor(). Would it be simpler to have a single color member that gets set by DetermineImportance (DetermineColor?) and then the `data.NomaiWallColor()` can be just a property access?

Reply:
Haven't the slightest idea of what you want here and how it would remotely improve the code, so leaving for you to change if you want.